src/Symfony/Bridge/Doctrine/Tests/Security/User/EntityUserProviderTest.php

```diff
@@ -233,14 +233,71 @@ private function getManager($em, $name = null)
 
     private function getObjectManager($repository)
     {
-        $em = $this->getMockBuilder(ObjectManager::class)
-            ->onlyMethods(['getClassMetadata', 'getRepository'])
-            ->getMockForAbstractClass();
-        $em->expects($this->any())
-            ->method('getRepository')
-            ->willReturn($repository);
-
-        return $em;
+        return new class($repository) implements ObjectManager {
+            private $repository;
+
+            public function __construct($repository)
+            {
+                $this->repository = $repository;
+            }
+
+            public function find($className, $id)
+            {
+            }
+
+            public function persist($object)
+            {
+            }
+
+            public function remove($object)
+            {
+            }
+
+            public function merge($object)
+            {
+            }
+
+            public function clear($objectName = null)
+            {
+            }
+
+            public function detach($object)
+            {
+            }
+
+            public function refresh($object)
+            {
+            }
+
+            public function flush()
+            {
+            }
+
+            public function getRepository($className)
+            {
+                return $this->repository;
+            }
+
+            public function getClassMetadata($className)
+            {
+            }
+
+            public function getMetadataFactory()
+            {
+            }
+
+            public function initializeObject($obj)
+            {
+            }
+
+            public function contains($object)
+            {
+            }
+
+            public function __call($name, $arguments)
+            {
+            }
+        };
```

Comment:
Why this complicated anonymous class? Shouldn't `$em = $this->createMock(ObjectManager::class);` have done the job?

Reply:
Updated 👍